src/streaming.ts

```diff
@@ -46,46 +52,116 @@ export class StreamProcessor {
-          'API-Key': this.apiKey,
-        },
+    this.log.info('Starting new StreamProcessor');
+
+    const url = `${this.options.baseUrl}/stream?cluster=${this.cluster}`;
+
+    const options = {
+      headers: {
+        'Cache-Control': 'no-cache',
+        Accept: 'text/event-stream',
+        Authorization: `Bearer ${this.jwtToken}`,
+        'API-Key': this.apiKey,
       },
-    );
+    };
 
-    eventSource.onopen = (event: MessageEvent) => {
-      this.log.debug('Stream connected', event);
+    const connected = () => {
+      this.log.info(`SSE stream connected OK: ${url}`);
+      this.retryAttempt = 1;
+      this.readyState = StreamProcessor.CONNECTED;
       this.eventBus.emit(StreamEvent.CONNECTED);
-      // In case we've had to retry, set it back to false
-      this.isRetrying = false
     };
 
-    eventSource.onretrying = (event: MessageEvent) => {
-      this.log.error('Stream is trying to reconnect', event);
-      // Only inform the client once that we're retrying
-      if (!this.isRetrying) {
-        this.isRetrying = true
-        this.eventBus.emit(StreamEvent.RETRYING);
-      }
-    };
+    const failed = (msg) => {
+      const delayMs = this.getRandomRetryDelayMs();
+      this.log.warn(`SSE disconnected: ${msg} will retry in ${delayMs}ms`);
+      this.readyState = StreamProcessor.RETRYING;
+      this.eventBus.emit(StreamEvent.RETRYING);
 
-    eventSource.onerror = (event: MessageEvent) => {
-      this.log.error('Unrecoverable error with stream, not retrying to connect: ', event);
-      this.eventBus.emit(StreamEvent.ERROR, event);
+      setTimeout(() => {
+        this.log.info('SSE retrying to connect');
+        this.connect(url, options, connected, failed);
+      }, delayMs);
     };
 
-    eventSource.addEventListener('*', (event: MessageEvent) => {
-      const msg: StreamMsg = JSON.parse(event.data);
+    this.connect(url, options, connected, failed);
 
-      this.log.debug('Received event from stream: ', msg);
+    this.eventBus.emit(StreamEvent.READY);
+  }
 
+  private getRandomRetryDelayMs(): number {
+    const delayMs = this.retryDelayMs * this.retryAttempt;
+    this.retryAttempt += 1;
+    return Math.min(delayMs, 60000);
+  }
+
+  private connect(
+    url: string,
+    options: RequestOptions,
+    connected,
+    failed,
+  ): void {
+    if (this.readyState === StreamProcessor.CONNECTED) {
+      this.log.debug('SSE already connected, skip retry');
+      return;
+    }
+
+    const isSecure = url.startsWith('https:');
+    this.log.debug('SSE HTTP start request', url);
+
+    (isSecure ? https : http)
+      .request(url, options, (res) => {
+        this.log.debug('SSE got HTTP response code', res.statusCode);
+
+        if (res.statusCode !== 200) {
+          failed('HTTP code ' + res.statusCode);
+          return;
+        }
+
+        connected();
+
+        res
+          .on('data', (data) => {
+            this.processData(data);
+          })
```

Comment:
Super minor and deffo not a blocker, but you can simplify a wee bit
```suggestion
          .on('data', this.processData)
```